src/Microsoft.Azure.WebJobs.Host/Executors/TriggeredFunctionData.cs

```diff
@@ -24,6 +24,6 @@ public class TriggeredFunctionData
         /// <summary>
         /// Optional handler function for processing the invocation.
         /// </summary>
-        public Func<Func<Task>, Task> InvokeHandler { get; set; }
+        public Func<Func<Task<object>>, Task<object>> InvokeHandler { get; set; }
```

Comment:
Since we intend to have these changes in 2.x, we can't make public API breaking changes.